src/drt/src/io/io.cpp

```diff
@@ -904,14 +905,49 @@ void io::Parser::setBTerms(odb::dbBlock* block)
                          box->getTechLayer()->getName());
         frLayerNum layerNum
             = tech_->name2layer[box->getTechLayer()->getName()]->getLayerNum();
-        frCoord xl = box->xMin();
-        frCoord yl = box->yMin();
-        frCoord xh = box->xMax();
-        frCoord yh = box->yMax();
+        frLayerNum finalLayerNum = layerNum;
+        odb::Rect viaBox;
+        bool useViaBox = false;
+        if (layerNum > TOP_ROUTING_LAYER) {
+          odb::dbNet* net = term->getNet();
+          odb::dbWire* wire = net->getWire();
+          if (wire != nullptr) {
+            odb::dbWirePath path;
+            odb::dbWirePathShape pshape;
+            odb::dbWirePathItr pitr;
+            for (pitr.begin(wire); pitr.getNextPath(path);) {
+              while (pitr.getNextShape(pshape)) {
+                if (pshape.shape.isVia()) {
+                  odb::dbTechVia* via = pshape.shape.getTechVia();
+                  for (const auto& vbox : via->getBoxes()) {
+                    layerNum = tech_->name2layer[vbox->getTechLayer()->getName()]
+                                   ->getLayerNum();
+                    if (layerNum == TOP_ROUTING_LAYER) {
+                      viaBox = vbox->getBox();
+                      odb::dbTransform xform;
+                      odb::Point path_origin = pshape.point;
+                      xform.setOffset({path_origin.x(), path_origin.y()});
+                      xform.setOrient(odb::dbOrientType(odb::dbOrientType::R0));
+                      xform.apply(viaBox);
+                      useViaBox = true;
+                      finalLayerNum = layerNum;
+                      termIn->setIsAboveTopLayer(true);
+                      break;
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+        frCoord xl = useViaBox ? viaBox.xMin() : box->xMin();
+        frCoord yl = useViaBox ? viaBox.yMin() : box->yMin();
+        frCoord xh = useViaBox ? viaBox.xMax() : box->xMax();
+        frCoord yh = useViaBox ? viaBox.yMax() : box->yMax();
```

Comment:
It seems simpler to just have a Rect to fill in rather than building it at the end:

```
Rect bbox = box->getBox();
...
if (...) {
  bbox = viaBox;
}
```